src/liballoc/boxed.rs

```diff
@@ -986,28 +986,40 @@ impl<E> Executor for Box<E>
 }
 
 #[unstable(feature = "futures_api", issue = "50547")]
-impl<'a, F: Future<Output = ()> + Send + 'a> From<PinBox<F>> for FutureObj<'a, ()> {
+impl<'a, F> From<PinBox<F>> for FutureObj<'a, F::Output>
```

Comment:
This is brilliant. I tried to add the `From` impl but could figure out how to do it. Now it seems so simple ^^'

Nit: It would be nice to use the type param naming convention `Fut` here

Reply:
Well, it would be so simple if rust didn't think it was incoherent. I've had this same issue a bunch of times before-- this *would* be a totally fine (if backwards-incompatible) impl, but the current set of coherence rules don't allow it.

Reply:
heh ^^'

with the upcoming `boxed()` combinator it's at least not so verbose anymore `future.boxed().into()` is almost as nice

Reply:
I just realized that my previous comment is nonsense. We would exactly this `From<PinBox>` impl...